src/Classes/PartyTab.lua

```diff
@@ -0,0 +1,620 @@
+	self.controls.importCodeState.label = function()
+		return self.importCodeDetail or ""
+	end
+	self.controls.importCodeMode = new("DropDownControl", {"TOPLEFT",self.controls.importCodeIn,"BOTTOMLEFT"}, 0, 4, 160, 20, { "All", "Aura", "Curse", "EnemyConditions", "EnemyMods" })
+	self.controls.importCodeMode.enabled = function()
+		return self.importCodeValid
+	end
+	self.controls.importCodeMode2 = new("DropDownControl", {"LEFT",self.controls.importCodeMode,"RIGHT"}, 8, 0, 160, 20, { "Replace", "Append", "Clear" })
+	self.controls.importCodeGo = new("ButtonControl", {"LEFT",self.controls.importCodeMode2,"RIGHT"}, 8, 0, 160, 20, "Import", function()
+		if self.controls.importCodeMode2.selIndex == 3 then
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 2 then
+				self.controls.editAuras:SetText("")
+				wipeTable(self.processedInput["Aura"])
+				self.processedInput["Aura"] = {}
+			end
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 3 then
+				self.controls.editCurses:SetText("")
+				wipeTable(self.processedInput["Curse"])
+				self.processedInput["Curse"] = {}
+			end
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 4 then
+				self.controls.enemyCond:SetText("")
+			end
+			if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 5 then
+				self.controls.enemyMods:SetText("")
+			end
+			wipeTable(self.enemyModList)
+			self.enemyModList = new("ModList")
+			self.build.buildFlag = true 
+			return
+		end
+		local importCodeFetching = false
+		if self.importCodeSite and not self.importCodeXML then
+			self.importCodeFetching = true
+			local selectedWebsite = buildSites.websiteList[self.importCodeSite]
+			buildSites.DownloadBuild(self.controls.importCodeIn.buf, selectedWebsite, function(isSuccess, data)
+				self.importCodeFetching = false
+				if not isSuccess then
+					self.importCodeDetail = colorCodes.NEGATIVE..data
+					self.importCodeValid = false
+				else
+					importCodeHandle(data)
+					finishImport()
+				end
+			end)
+			return
+		end
+
+		finishImport()
+	end)
+	self.controls.importCodeGo.label = function()
+		return self.controls.importCodeMode2.selIndex == 3 and "Clear" or "Import"
+	end
+	self.controls.importCodeGo.enabled = function()
+		return (self.importCodeValid and not self.importCodeFetching) or self.controls.importCodeMode2.selIndex == 3
+	end
+	self.controls.importCodeGo.x = function()
+		return (self.width > 1350) and 8 or (-328)
+	end
+	self.controls.importCodeGo.y = function()
+		return (self.width > 1350) and 0 or 28
+	end
+	self.controls.importCodeGo.enterFunc = function()
+		if self.importCodeValid then
+			self.controls.importCodeGo.onClick()
+		end
+	end
+	
+	self.controls.rebuild = new("ButtonControl", {"LEFT",self.controls.importCodeGo,"RIGHT"}, 8, 0, 160, 20, "Rebuild All", function() 
```

Comment:
Any reason this isn't just being done whenever you edit/import text?  Much like the custom mods box.

Reply:
I see it's somewhat useful for the "Remove effects" feature, but I think that could be done via a checkbox instead.

Reply:
it breaks if you change specific things, or if people type it out, I can make it auto rebuild when focus is lost, but it can cause temporary issues if its rebuilt every time the "buf" changes (though I fixed all the known crashes so the issues are likly just minor temporary issues untill the full form is typed out)